plugins/package-managers/spm/src/main/kotlin/SpmModel.kt

```diff
@@ -92,6 +74,21 @@ data class Pin(
     )
 }
 
+internal val LibraryDependency.id: Identifier
+    get() = Identifier(
+        type = PACKAGE_TYPE,
+        namespace = "",
+        name = getCanonicalName(repositoryUrl),
+        version = version
+    )
+
+internal fun LibraryDependency.toPackage(): Package {
+    val vcsInfoFromUrl = VcsHost.parseUrl(repositoryUrl)
+    val vcsInfo = vcsInfoFromUrl.takeUnless { it.revision.isBlank() } ?: vcsInfoFromUrl.copy(revision = version)
```

Comment:
Nit: The "While at it" part of the commit message could also mention that the `vcs` property was inlined.